parson.c

```diff
@@ -698,8 +698,9 @@ static _Nt_array_ptr<char> process_string(_Nt_array_ptr<const char> input : coun
     _Nt_array_ptr<const char> input_ptr : bounds(input, input + len) = input;
     size_t initial_size = (len + 1) * sizeof(char);
     size_t final_size = 0;
-    _Nt_array_ptr<char> output : count(initial_size) = NULL, output_ptr : bounds(output, output + initial_size) = NULL;
+    _Nt_array_ptr<char> output : count(initial_size) = NULL;
     output = parson_string_malloc(initial_size);
+    _Nt_array_ptr<char> output_ptr : bounds(output, output + initial_size) = NULL;
```

Comment:
The reason that moving the declaration of `output_ptr` fixes the compilation error is this:

The declared bounds `(output, output + initial_size)` of `output_ptr` use the value of `output`. In the assignment `output = parson_string_malloc(initial_size)` on line 702, the expression `parson_string_malloc(initial_size)` is not invertible with respect to `output`. If `output_ptr` is declared before line `702` with a value of `NULL`, then the original value of `output` in line 702 is `output_ptr` (since `output` and `output_ptr` have the same value at that point). This means that the inferred bounds of `output_ptr` at the end of checking line 702 are `(output_ptr, output_ptr + initial_size)`. At that point, there is no relationship between `output` and `output_ptr`, so the compiler cannot prove that the inferred bounds `(output_ptr, output_ptr + initial_size)` of `output_ptr` imply its declared bounds `(output, output + initial_size)`. Therefore, `output_ptr` should be declared after the assignment on line 702.

Reply:
Thank you for the explanation!.